### DISCUSSION TITLE
How to measure time between content item events?

### DISCUSSION THREAD

**Discussion by intellimedhu** [OrchardCMS/OrchardCore#11772] [Q&A] [answered]
I want to measure the time between content item events. My first approach is to use `ContentHandler`. For example, it would be nice to know how long it takes from the start of creating a content item to its publication: `timediff(InitializingAsync, PublishedAsync)`. The problem is that `_contentManager.NewAsync` is called more than once. In the admin content item list view, it is 3x for permission checking and other reasons ([see here](https://github.com/OrchardCMS/OrchardCore/blob/f689eb8da7d24c0fead608ac0e2d8e829914ca14/src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs#L136)) but isn't used anywhere.

What would be your suggestion?

**Comment by Skrypt** [comment]
This is not really creating a content item. It is just creating a "fake" content item of a certain content type to analyze if the user has permission. It should not take much time to just create the empty content item.

**Reply by intellimedhu** [reply]
I know most of the content items are "fake", but the one I can see in the editor is "real". How can I be sure which one is that?

**Comment by ns8482e** [comment]
@intellimedhu for `New` you really tap into DraftSaving event

**Reply by intellimedhu** [reply]
`Published` event was only an example. Probably most of the events will be relevant.

**Comment by intellimedhu** [comment, marked as answer]
I'll check the request path `$"/Contents/ContentTypes/{contentType/Create"`. If matching "only" 3 fake content items were created, the 4th one is what I need so I can store the events. Creating a "cleanup" method is a possible solution.
